glean-core/src/storage/mod.rs

```diff
@@ -42,6 +42,33 @@ fn snapshot_labeled_metrics(
     obj.insert(label.into(), metric.as_json());
 }
 
+/// Labeled metrics are stored as `<metric id>/<label>`.
+/// They need to go into a nested object in the final snapshot.
+///
+/// We therefore extract the metric id and the label from the key and construct the new object or
+/// add to it.
+fn snapshot_dual_labeled_metrics(
+    snapshot: &mut HashMap<String, HashMap<String, JsonValue>>,
+    metric_id: &str,
+    metric: &Metric,
+) {
+    let ping_section = format!("dual_labeled_{}", metric.ping_section());
+    let map = snapshot.entry(ping_section).or_default();
+
+    // Safe unwrap, the function is only called when the id does contain a '/'
+    let parts = metric_id.split('\x1e').collect::<Vec<&str>>();
```

Comment:
Might be good to reuse the defined constant. But that requires to make it `pub(crate)` over in `glean-core/src/metrics/dual_labeled_counter.rs` and then also a `use` statement.